src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -6548,14 +6552,20 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
 
         if (!s) {
             std::set<std::string> not_cpython_builtin = {
-                "sin", "cos", "gamma", "tan", "asin", "acos", "atan"
+                "sin", "cos", "gamma", "tan", "asin", "acos", "atan", "exp", "exp2"
             };
-            if (ASRUtils::IntrinsicFunctionRegistry::is_intrinsic_function(call_name)
-             && not_cpython_builtin.find(call_name) == not_cpython_builtin.end()) {
+            if (ASRUtils::IntrinsicFunctionRegistry::is_intrinsic_function(call_name) &&
+                (not_cpython_builtin.find(call_name) == not_cpython_builtin.end() ||
+                imported_functions.find(call_name) != imported_functions.end() )) {
                 ASRUtils::create_intrinsic_function create_func =
                     ASRUtils::IntrinsicFunctionRegistry::get_create_function(call_name);
                 Vec<ASR::expr_t*> args_; args_.reserve(al, x.n_args);
                 visit_expr_list(x.m_args, x.n_args, args_);
+                if (ASRUtils::is_array(ASRUtils::expr_type(args_[0])) && 
+                    current_scope->resolve_symbol("numpy") == nullptr ) {
```

Comment:
this seems to be the wrong check, I think it doesn't matter if `import numpy` is present, but rather if `exp` is imported from numpy.

Reply:
@czgdp1807 we already have this issue with `sin` and other functions I think, so we can probably fix this in a separate PR.